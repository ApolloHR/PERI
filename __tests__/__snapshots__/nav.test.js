import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import { Link } from 'react-router-dom';

import Nav from '../../client/src/components/nav.jsx';

describe('>>>N A V --- Test nav component', () => {

  it('renders home link', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <Link to="/" className="has-text-grey-dark"activeClassName="active" > Home </Link>
    </MemoryRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
  });

  it('renders Build Trip link', () => {
  const tree = renderer
    .create(<MemoryRouter>
      <Link to="/buildTrip" className="has-text-grey-dark"activeClassName="active" > Build Trip </Link>
    </MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
  });

  it('renders Cart link', () => {
  const tree = renderer
    .create(<MemoryRouter>
      <Link to="/cart" className="has-text-grey-dark"activeClassName="active" > Cart </Link>
    </MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
  });

  it('renders G-login link', () => {
  const tree = renderer
    .create(<MemoryRouter>
      <Link to="/auth/google" className="has-text-grey-dark"activeClassName="active" > G-login </Link>
    </MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
  });

  it('renders FB-Login link', () => {
  const tree = renderer
    .create(<MemoryRouter>
      <Link to="/auth/facebook" className="has-text-grey-dark"activeClassName="active" > FB-login </Link>
    </MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
  });
});