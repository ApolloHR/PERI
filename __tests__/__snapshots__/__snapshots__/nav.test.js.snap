// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>N A V --- Test nav component renders Build Trip link 1`] = `
<a
  activeClassName="active"
  className="has-text-grey-dark"
  href="/buildTrip"
  onClick={[Function]}
>
   Build Trip 
</a>
`;

exports[`>>>N A V --- Test nav component renders Cart link 1`] = `
<a
  activeClassName="active"
  className="has-text-grey-dark"
  href="/cart"
  onClick={[Function]}
>
   Cart 
</a>
`;

exports[`>>>N A V --- Test nav component renders FB-Login link 1`] = `
<a
  activeClassName="active"
  className="has-text-grey-dark"
  href="/auth/facebook"
  onClick={[Function]}
>
   FB-login 
</a>
`;

exports[`>>>N A V --- Test nav component renders G-login link 1`] = `
<a
  activeClassName="active"
  className="has-text-grey-dark"
  href="/auth/google"
  onClick={[Function]}
>
   G-login 
</a>
`;

exports[`>>>N A V --- Test nav component renders home link 1`] = `
<a
  activeClassName="active"
  className="has-text-grey-dark"
  href="/"
  onClick={[Function]}
>
   Home 
</a>
`;
